demo_modules/penrose_tiles/penrose_tiles_client.ts

```diff
@@ -38,54 +43,57 @@ export function load(
 
       this.tilesState = state.define("tiles", {
         currentGeneration: CurrentValueInterpolator,
-        tileGenerations: [[
-          {
-            points: CurrentValueInterpolator,
-            angle: CurrentValueInterpolator,
-            size: CurrentValueInterpolator,
-            type: CurrentValueInterpolator,
-            extents: CurrentValueInterpolator,
-          },
-        ]],
         kiteHue: NumberLerpInterpolator,
+        kiteSat: NumberLerpInterpolator,
+        kiteLgt: NumberLerpInterpolator,
         dartHue: NumberLerpInterpolator,
+        dartSat: NumberLerpInterpolator,
+        dartLgt: NumberLerpInterpolator,
       });
-    }
 
-    beginFadeIn(_time: number) {}
+      const center = wallGeometry.extents.center();
+
+      let lastGen = Tile.protoTiles(center, wallGeometry.extents.w / 2.5);
 
-    finishFadeIn() {}
+      this.generations[0] = this.precomputeTiles(lastGen);
+
+      // Compute the rest of the tiles in the background
```

Comment:
What happens if the generation n+1 isn't done by the time it's supposed to transition to it?

From what I understand this shouldn't theoretically be an issue given the 2nd gen will be very simple and then each subsequent one gets harder but also has more time.

Reply:
Yeah - each generation will be displayed for 20 seconds, so that should be ample time by the time it gets to the later generations for them to be done. If they aren't for some reason, it'll probably crash. :)

Reply:
Also, each client only stores (temporarily) the tiles that it can see, so it shouldn't be too much of a memory burden - each gen generates _all_ the tiles, but only the previous generation is kept (temporarily) until the new one is calculated.